src/hooks/use-ads.tsx

```diff
@@ -0,0 +1,60 @@
+import type { MutableRefObject } from 'react';
+import { useEffect } from 'react';
+
+const SCRIPT_URL
+  = '//cdn.carbonads.com/carbon.js?serve=CEAI627Y&placement=sequelizeorg';
+
+type OnEnvironment = 'mobile' | 'desktop' | 'all';
+
+type InitProps = {
+  ref?: MutableRefObject<HTMLInputElement | undefined>,
+  selector?: string,
+  on?: OnEnvironment,
+};
+
+function shouldRender(on: OnEnvironment) {
+  const isMobile = window.matchMedia('(max-width: 996px)').matches;
+
+  if (on === 'all') {
+    return true;
+  }
+
+  if (on === 'mobile') {
+    return isMobile;
+  }
+
+  return !isMobile;
+}
+
+export function useAds({ ref, selector, on = 'all' }: InitProps): void {
+  useEffect(() => {
+    let container: HTMLElement | null = null;
+    let script: HTMLScriptElement | null = null;
+
+    if (shouldRender(on)) {
+      if (ref) {
+        container = ref.current as any;
+      } else if (selector) {
+        container = document.querySelector(selector);
+      }
+    }
+
+    if (container) {
+      script = document.createElement('script');
+
+      script.id = '_carbonads_js';
+      script.src = SCRIPT_URL;
+      script.async = true;
+
+      container.append(script);
+    }
+
+    return () => {
+      if (!container) {
+        return;
+      }
+
+      script?.remove();
```

Comment:
The thing about scripts is that even if you remove the script, it will still continue to execute. Removing the tag has no effect.

By resizing the screen, you can end up with this:

![image](https://user-images.githubusercontent.com/1280915/166144377-5bbf646f-bcde-43cf-9aab-66887320a7f6.png)

Reply:
should be fixed